### PR TITLE
fix: evaluation event being reported with no user ID

### PR DESCRIPTION
**Pull request by hvn2k1**
To resolve https://github.com/bucketeer-io/go-server-sdk/issues/148

### CODE REVIEW

File: pkg/bucketeer/sdk.go [hunk 1]
```diff
@@ -347,6 +347,22 @@ func getEvaluationDetails[T model.EvaluationValue](
 ) model.BKTEvaluationDetails[T] {
 	var err error
 	var value T
+
+	err = validateGetEvaluationRequest(user, featureID)
```
Comment (cre8ivejp) [inline pkg/bucketeer/sdk.go:351]: We also need the validation here.
https://github.com/bucketeer-io/go-server-sdk/blob/afc407461536bfe795d8d7c952c7970ec4f114ae/pkg/bucketeer/sdk.go#L467-L471

Comment (hvn2k1) [reply pkg/bucketeer/sdk.go:351]: I have added the validation

File: pkg/bucketeer/sdk.go [hunk 2]
```diff
@@ -347,6 +347,21 @@ func getEvaluationDetails[T model.EvaluationValue](
...
+		s.logVariationError(err, logFuncName, user.ID, featureID)
+		return model.NewEvaluationDetails[T](
+			featureID,
+			user.ID,
+			"",
+			"",
+			0,
+			model.ReasonClient,
+			defaultValue,
+		)
```
Comment (cre8ivejp) [inline pkg/bucketeer/sdk.go:362]:
```suggestion
		s.logVariationError(err, logFuncName, user.ID, featureID)
		var userID string
		if user.Valid() {
		  userID = user.ID
		}
		return model.NewEvaluationDetails[T](
			featureID,
			userID,
			"",
			"",
			0,
			model.ReasonClient,
			defaultValue,
		)
```

**Review by cre8ivejp — APPROVED** [review]
Nice!
Thank you!

**Pull request merged**